### PR TITLE
Remote lambda run config UI bug fix

### PR DESCRIPTION
**Pull request by manodnyab**
## Types of changes

- [ X ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description

The lambda function requested is displayed in function selector of Remote Lambda Run Configuration UI even if it isn't in the active region. The functions that can be selected correspond to connection settings configured in the remote lambda run config UI. The connection settings changed listener is now invoked every time the connection settings are changed (without explicit user interaction) , which in turn reloads the functions that can be chosen according to specified credentials/region.

## Motivation and Context

Remote lambda run config UI broke if the requested lambda was not in the active region. The lambda functions corresponding to the AWS connection settings configured in the Remote Lambda Run Config UI are now correctly displayed.

## Checklist

- [ x ] I have read the **README** document
- [ x ] I have read the **CONTRIBUTING** document
- [ ] Local run of `gradlew check` succeeds
- [ x ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ x ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/ui/connection/AwsConnectionSettingsSelector.kt [hunk 1]
```diff
@@ -91,6 +90,7 @@ class AwsConnectionSettingsSelector(
         } catch (_: Exception) {
             view.credentialProvider.setSelectedInvalidCredentialsProvider(credentialProviderId)
         }
+        fireChange()
```
Comment (hunterwerlla) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/ui/connection/AwsConnectionSettingsSelector.kt:93]: How does this actually do anything? The listener is empty in LocalLambdaRunConfiguration?

Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/ui/connection/AwsConnectionSettingsSelector.kt:93]: Ah I see, I thought this was a different bug

File: .changes/next-release/bugfix-98fc2305-2aec-4bbd-974a-4bcaff6f1622.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "type" : "bugfix",
+  "description" : "Remote Lambda Run Config UI now displays correct functions in accordance with the AWS connection settings configured in the UI."
```
Comment (hunterwerlla) [inline .changes/next-release/bugfix-98fc2305-2aec-4bbd-974a-4bcaff6f1622.json:3]: This doesn't really describe what was fixed

Comment (manodnyab) [reply .changes/next-release/bugfix-98fc2305-2aec-4bbd-974a-4bcaff6f1622.json:3]: Would this be a better description? `Function selector in the Remote Lambda Run Config UI is now filled with requested lambda name even if it isn't in the active region.`

Comment (hunterwerlla) [reply .changes/next-release/bugfix-98fc2305-2aec-4bbd-974a-4bcaff6f1622.json:3]: Maybe: `Fixed the Remote Lambda Run Configuration periodically failing to load the list of functions`?

Comment (manodnyab) [reply .changes/next-release/bugfix-98fc2305-2aec-4bbd-974a-4bcaff6f1622.json]: Okay, got the difference between the two! Changed it in the next commit

**Review by abrooksv — APPROVED** [review]

**Pull request merged**